beep/protocol_tools/arbin_schedule_file.py

```diff
@@ -0,0 +1,251 @@
+# Copyright 2019 Toyota Research Institute. All rights reserved.
+""" Schedule file parsing and parameter insertion"""
+
+
+import os
+import hashlib
+import re
+from beep import SCHEDULE_TEMPLATE_DIR
+from collections import OrderedDict, defaultdict
+
+
+class ScheduleFile:
+    """
+    Schedule file utility. Provides the ability to read an Arbin type schedule file
+
+    Args:
+        version (str): Default version.
+        section_regex (raw str): regex string to return all section headers from
+            the schedule file
+        step_regex (raw str): regex string to return all step headers from the
+            schedule file
+        limit_regex (raw str): regex string to return all limit headers from
+            the schedule file
+
+    """
+    def __init__(self,
+                 version=None,
+                 section_regex=r'(?<=\[).*',
+                 step_regex=r'.Schedule_Step[0-9]*',
+                 limit_regex=r'.Schedule_Step[0-9]*_Limit[0-9]*'):
+        self.service = version
+        self.section = section_regex
+        self.step = step_regex
+        self.limit = limit_regex
+        self.schedule_file_encoding = 'latin-1'
+
+    def hash_file(self, inputfile):
+        with open(inputfile, 'rb') as f:
+            chunk = f.read()
+        return hashlib.md5(chunk).digest()
+
+    def to_dict(self, inputfile):
+        """Schedule file ingestion. Converts a schedule file with section headers
+        to an ordered dict with section headers as nested dicts. One line in the
+        schedule file is not parsable by utf-8. This line is stored and returned
+        separately with the line number that it came from
```

Comment:
It seems that the non-parseable line is not stored in this version? It also no longer seems to be part of dict_to_file.

Reply:
Yeah, I was able to change the encoding to read that line so I will update the doc string. Thanks for spotting that!